### ISSUE TITLE
Alert doesn't support usePortal and portalClassName TS definitions, but they work fine

### ISSUE THREAD

**Issue by shravan2x** [palantir/blueprint#5801]
#### Environment

- __Package version(s)__: 4.13.0
- __Operating System__: Windows 11
- __Browser name and version__: Chrome latest

#### Steps to reproduce

Use the `usePortal` prop on a Blueprint TSX project. A red error message displays the message ` Property 'usePortal' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Alert> & Pick<Readonly<IAlertProps>, never> & InexactPartial<...> & InexactPartial<...>'.`.

The Blueprint docs at https://blueprintjs.com/docs/#core/components/alert and https://github.com/palantir/blueprint/blob/develop/packages/core/src/components/alert/alert.tsx don't mention these two props either. But forcing compilation works and it's clear from DevTools that usePortal works fine.

**Comment by adidahiya** [comment]
Yeah, we've restricted the "overlayable props" that type-check and are officially supported by the `<Alert>` component because we want to be more prescriptive about how that component is used compared to a `<Dialog>`.

I'm reluctant to add support for `<Alert usePortal={false}>`. Showing an Alert outside of a portal, inline in some UI, would defeat some of the design constraints and standardization of this component in our design system. Why you need this prop? What is your use case for not using a portal?

`<Alert>` is a relatively simple wrapper around `<Dialog>`, so I would probably suggest you create a custom `<Dialog>` to achieve what you're trying to do here.

**Comment by shravan2x** [comment]
In my case, I don't want alerts blocking the view of the Navbar. The rest of the page can be covered by the overlay.

Copying all Alert logic over to Dialog just for this seems like overkill. I suppose I could suppress the TS error, but I suggest we consider supporting this use-case. A navbar is a critical component that I would expect to always be visible.

**Comment by adidahiya** [comment]
Alright, I'll take a PR for this. The new prop would to come with a strong warning, `Not recommended. Only for use in rare circumstances.`
